11-bug_tracking.Rmd

```diff
@@ -1,23 +1,125 @@
+
+### Bug in the $\textsf{R}$ Language {#RLangBug}
+
+1. If there is a bug in the $\textsf{R}$ language itself, then you need to report using the same procedure and precautions as explained [above](#RCorePkgBug).
+
+### Bug in the non $\textsf{R}$ Core-supported packages and/or their documentations {#nonRCorePkgBug}
+
+1. For packages that are not maintained by the $\textsf{R}$ Core, the bug reports are usually submitted to their maintainer(s). You can find who is the package maintainer by running either of the commands `maintainer("package_name")` or `packageDescription("package_name")`.
+
+2. In $\textsf{R}$ the running function `bug.report(package = "package_name")` shall direct you to either the GitHub issue tracker of the package, or to the bug tracking web page, or towards composing an e-mail to the package maintainer. This function `bug.report` is disabled in $\textbf{RStudio}$.
+
+3. Please ensure that your bug is not already reported or fixed before reporting it in any of the ways suggested above.
+
+## Good practices in reporting bugs / Expectations of a (good) bug report
+
+If you follow the practices given below, you will come up with a good bug report which might make it easier for the Maintainer(s) to fix the bug.
+
+1. Include a minimal reproducible example of the bug in your report. The maintainer should be able to quickly reproduce the bug on using the minimal example that you provide.
+
+2. Mention the software architecture on which the bug occurred. 
+
+In addition to the above, here are the [bug writing guidelines](https://bugs.r-project.org/bugzilla/page.cgi?id=bug-writing.html) on Bugzilla. The [bug reporting](https://www.r-project.org/bugs.html#writing-a-good-bug-report) documentation in R also the discusses practices to write a good bug report.
+
+Once you have successfully reported a bug, you will likely receive an update each time an action is taken on the bug. On Bugzilla, the report may be given one of the following status: New, Assigned, Confirmed, Reopened, Unconfirmed.
+

```

Comment:
When I first submitted a bug I didn't know the Severity to give to a bug or the Component to use, perhaps some paragraphs about each element of a bug report will make it easier for people to understand how to make a good bug report on Bugzilla

Reply:
Yes. This would be good to discuss e.g. at the RCWG and get input from R core members.

Reply:
Yes. This would be good to discuss e.g. at the RCWG and get input from R core members.